qiskit/algorithms/eigensolvers/eigensolver.py

```diff
@@ -0,0 +1,106 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The eigensolver interface and result."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any
+import numpy as np
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from ..algorithm_result import AlgorithmResult
+from ..list_or_dict import ListOrDict
+
+
+class Eigensolver(ABC):
+    """The eigensolver interface.
+
+    Algorithms that can compute eigenvalues for an operator
+    may implement this interface to allow different algorithms to be
+    used interchangeably.
+    """
+
+    @abstractmethod
+    def compute_eigenvalues(
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None = None,
+    ) -> "EigensolverResult":
+        """
+        Computes eigenvalues. The ``operator`` and ``aux_operators`` can be supplied here and,
+        if not ``None``, will override any already set into algorithm so it can be reused with
```

Comment:
This text about if not None will override any already set goes back to a time when the operator could also be passed in via the constructor and maybe there was setter as well. Whatever the history of why this is so this text is no longer relevant. I check the MinimumEigensolver and it says something very similar so may I can ask that this be updated at the same time. Basically you supply an main operator here and optional aux operators. There is no remove a previous etc etc. This call is now standalone.